google/ggrc-core
# GGRC-8131 Line Break Preservation in GGRC

# Issue description

Line breaks are not being preserved in GGRC during import-export from UI to gdrive and back. When exporting data (assessments) from UI to gdrive, text in rich-text columns (test_plan, notes and description) are exported with html tags and is unreadable to user.

# Steps to test the changes

1. In dev branch: create assessment, in Assessment Procedure type 1 2 3 with empty line between the chars.
2. Export assessment, open it in gdrive. Expected text:
  `<p>1</p><p>2</p><p>3</p>`
3. Without changes, import this assessment via UI and open it. Expected – 1 2 3 in one line, with spaces between the chars.
4. In GGRC-8131 branch: do the same, expected: in gdrive – text without tags, every char in new line. Expected after import – the same as before import, every char in separate line.
5. Check statuses: create assessment, in Assessment procedure type

# Solution description

In order to preserve line breaks (i.e. make text in these columns human-readable) new RichTextColumnHandler was created. In it `<p>` and `<br>` tags are sanitized and replaced with newline character while export, and newline character replaced with `<p>{}</p>` while import via UI.

The logic for auto status change was also corrected – for rich-text fields that was modified by new column handler, new logic applied. For assessments in DONE_STATE and FINAL_STATE, text in these columns is compared before and after commit stripped off `<p>`, `<br>` and `\n`. So, if text is the same before and after import of assessment, status wouldn’t be changed.

Possible problems: If user wants to format text using empty lines between text and waits that status would be changed to In Progress, this solution could be a problem, because after adding new lines the text would be the same after comparison, and status would be preserved uchanged.

Also changes should be checked with GGRCQ, because of import-export operation and sync.


# Sanity checklist




# PR Review checklist
